src/lycophron/tasks/tasks.py

```diff
@@ -6,218 +6,143 @@
-        return record
-
-    publish_url = record.links.get("publish")
-
-    if not publish_url:
-        print(f"No publish url provided for record {record.doi or record.id}")
-        return
-
-    if record.status not in (
-        RecordStatus.PUBLISH_FAIL,
-        RecordStatus.FILE_SUCCESS,
-        RecordStatus.DEPOSIT_SUCCESS,
-    ):
-        print(
-            f"Record {record.doi or record.id} status {record.status}: record not published."
-        )
-        return
-
-    session = create_session(token)
-    r = session.post(publish_url)
-    if r.status_code != 202:
-        print(
-            f"Publish for record {record.doi or record.id} failed with code {r.status_code} text {r.text}"
-        )
-        _publish_failed(record, r.text)
-    else:
-        data = r.json()
-        html_url = data["links"]["record_html"]
-        _publish_succeeded(record, data["metadata"])
-        print(f"Record published. Find it in : {html_url}")
-
-
-def upload_files(record, token):
-    """Upload a file to a deposit."""
-
-    def _upload_success(record):
-        record.status = RecordStatus.FILE_SUCCESS
-        db.update_record(record)
-        return record
-
-    def _upload_failed(record, text=None):
-        record.status = RecordStatus.FILE_FAIL
-        if text:
-            record.response = text
-        db.update_record(record)
-        return record
-
-    session = create_session(token)
-    bucket_url = record.links.get("bucket")
-    if not bucket_url:
-        print(
-            f"Record {record.doi or record.id} failed to upload files. No bucket url found."
-        )
-        return
-
-    if record.status not in (
-        RecordStatus.FILE_FAIL,
-        RecordStatus.DEPOSIT_SUCCESS,
-    ):
-        print(
-            f"Record {record.doi or record.id} status {record.status}: files can't be uploaded."
-        )
-        return
-
-    for file in record.files:
-        file_name = file["filename"]
-        filepath = file["filepath"]
-        full_path = os.path.join(os.getcwd(), filepath)
-
-        failed = False
-        with open(full_path, "rb") as fp:
-            r = session.put(
-                "%s/%s" % (bucket_url, file_name),
-                data=fp,
-                params={},
-                headers={"Content-Type": "application/octet-stream"},
-            )
-            if r.status_code != 200:
-                failed = True
-
-        if failed:
-            _upload_failed(record)
-        else:
-            _upload_success(record)
-
-        fp.close()
+        if record.status == RecordStatus.TODO:
+            record.status = RecordStatus.QUEUED
+            db.session.commit()
+        elif record.error:
+            continue # TODO Reporting error well (Awaiting fixes in client)
+        elif record.response:
+            continue # TODO Manage 5xx/4xx
+        elif (datetime.utcnow() - record.updated).seconds < 180:
+            continue
+
+        process_record.apply_async(args=[record.id])
```

Comment:
Major: In this case, would it be possible that if - for whatever reason - a record is stuck being processed for a long time, let's say the http request are slow, and there is a queue of records to be processed if we don't change the state as soon as they are picked up by the celery task we risk that there will be another task that will pick them and try to process them as well,  I would suggest that we first update the status of all the records to avoid potential issues:
```suggestion
    lapp = LycophronApp()
    db = lapp.project.db

    # First, update the status of all TODO records to QUEUED
    records = db.get_unpublished_deposits(num_records)
    for record in records:
        if record.status == RecordStatus.TODO:
            record.status = RecordStatus.QUEUED

    # Commit the status updates to the database
    db.session.commit()
    
    # Process each record
    .... 
```

Reply:
That's exactly the reason this `QUEUED` state was added. 
```python
process_record.apply_async(args=[record.id])
```
just adds the tasks to the celery queue and doesn't wait for it to finish, so there is no bottleneck on http requests. We can we sure that these records will be moved to QUEUED before they get picked up unless the db fails to commit, which is unlikely in our case.

We have also addressed the case of multiple same records in the queue. The only possibility here is a record gets queued twice but the minimum difference between them will be 180 seconds, which we optimistically assume will be enough for the first iteration.

Reply:
RIght! DIdn't realize the record is processed inside a celery task :see_no_evil: , in that case there is no real risk of having the same record being processed twice.

However, the suggested code will improve a bit in terms of efficiency as it aggregates all the changes and then commits them in a single transaction. This reduces the number of write operations to the database, in this case it's not very imporant, but it's something we need to keep in mind in general as in RDM in many cases we abuse on the number of queries we do! :upside_down_face: